File: vue.jsonforms/src/entities/client-money/InsuranceProvider.ts
import { ContactNumber } from "@/entities/ContactNumber";

export class InsuranceProvider {
  public providerName = "";
  public companyNumber: number | undefined;
  public firmReferenceNumber: number | undefined;
  public isPraAuthorized = false;
  public insurerRating: string | undefined;
  public registeredAddress: string | undefined;
  public tradingAddress: string | undefined;
  public emailAddress: string | undefined;
  public contactNumber: ContactNumber | undefined;
  public website: string | undefined;
  public withBindingAuthority: boolean | undefined;
  public haveRiskTransferAgreement: boolean | undefined;
  public allowToCascadeRiskTransferAgreement: boolean | undefined;
}

File: CSharp.Functions/Common/Entities/FirmDetails/ClientMoney/InsuranceProvider.cs
namespace Common.Entities
{
    public class InsuranceProvider
    {
        public string? ProviderName { get; set; }
        public int CompanyNumber { get; set; }
        public int FirmReferenceNumber { get; set; }
        public bool IsPraAuthorized { get; set; }
        public string? InsurerRating { get; set; }
        public string? RegisteredAddress { get; set; }
        public string? TradingAddress { get; set; }
        public string? EmailAddress { get; set; }
        public ContactNumber? ContactNumber { get; set; }
        public string? Website { get; set; }
        public bool WithBindingAuthority { get; set; }
        public bool HaveRiskTransferAgreement { get; set; }
        public bool AllowToCascadeRiskTransferAgreement { get; set; }
    }
}
